paltas/Configs/config_handler.py

```diff
@@ -0,0 +1,621 @@
+			sub_model_list, sub_kwargs_list, sub_z_list = (
+				self.subhalo_class.draw_subhalos())
+			complete_lens_model_list += sub_model_list
+			complete_lens_model_kwargs += sub_kwargs_list
+			complete_z_list += sub_z_list
+		if self.main_deflector_class is not None:
+			self.main_deflector_class.update_parameters(
+				sample['main_deflector_parameters'],sample['cosmology_parameters'])
+			main_model_list, main_kwargs_list, main_z_list = (
+				self.main_deflector_class.draw_main_deflector())
+			complete_lens_model_list += main_model_list
+			complete_lens_model_kwargs += main_kwargs_list
+			complete_z_list += main_z_list
+		if self.lens_light_class is not None:
+			self.lens_light_class.update_parameters(
+				cosmology_parameters=sample['cosmology_parameters'],
+				source_parameters=sample['lens_light_parameters'])
+			lens_light_model_list, lens_light_kwargs_list = (
+				self.lens_light_class.draw_source())
+		if self.point_source_class is not None:
+			self.point_source_class.update_parameters(sample['point_source_parameters'])
+			point_source_model_list,point_source_kwargs_list = (
+				self.point_source_class.draw_point_source())
+
+		# Now get the model and kwargs from the source class (which must
+		# always be present.)
+		self.source_class.update_parameters(
+			cosmology_parameters=sample['cosmology_parameters'],
+			source_parameters=sample['source_parameters'])
+
+		# For catalog objects we also want to save the catalog index
+		# and the (possibly randomized) additional rotation angle. We will
+		# therefore push these back into the sample object.
+		if isinstance(self.source_class,GalaxyCatalog):
+			catalog_i, phi = self.source_class.fill_catalog_i_phi_defaults()
+			source_model_list, source_kwargs_list = self.source_class.draw_source(
+				catalog_i=catalog_i, phi=phi)
+			sample['source_parameters']['catalog_i'] = catalog_i
+			sample['source_parameters']['phi'] = phi
+		else:
+			source_model_list, source_kwargs_list = (
+				self.source_class.draw_source())
+
+		# Populate the source redshift as a list for concistency
+		source_redshift_list = [sample['source_parameters']['z_source']]*len(
+			source_model_list)
+
+		# Package all of the lists into a dict
+		lenstronomy_dict = {}
+		lenstronomy_dict['lens_model_list'] = complete_lens_model_list
+		lenstronomy_dict['lens_kwargs_list'] = complete_lens_model_kwargs
+		lenstronomy_dict['lens_redshift_list'] = complete_z_list
+		lenstronomy_dict['lens_light_model_list'] = lens_light_model_list
+		lenstronomy_dict['lens_light_kwargs_list'] = lens_light_kwargs_list
+		lenstronomy_dict['point_source_model_list'] = point_source_model_list
+		lenstronomy_dict['point_source_kwargs_list'] = point_source_kwargs_list
+		lenstronomy_dict['source_model_list'] = source_model_list
+		lenstronomy_dict['source_kwargs_list'] = source_kwargs_list
+		lenstronomy_dict['source_redshift_list'] = source_redshift_list
+
+		return lenstronomy_dict
```

Comment:
Is it possible that the return is split in two keyword arguments, one defining the model type (such as kwargs_model with 'lens_model_list' etc in standard lenstronomy conventions and one as kwargs_params with 'kwargs_lens', 'kwargs_source' etc? Otherwise the user needs to split them manually afterwards before being used

Reply:
I can definitely split them up, although there's a third type of list in this dictionary, the list of lens / source redshifts. I would have to break it up into three dictionaries, one containing the models, one containing the kwargs, and one containing the redshifts. Would that still be better? Happy to do it: it's a very easy change to make.

Reply:
The lens and source redshift are also in kwargs_model based on the current lenstronomy conventions. There are only kwargs_model and kwargs_param that together are able to fully describe a lenstronomy model. Or am I missing something?

Reply:
Oh I see, this is in terms of the inputs to ModelAPI in lenstronomy. I'll make the change!